### PR TITLE
Remove interface compatibility hollow sphere

### PR DESCRIPTION
**Pull request by alarshi**
In benchmarks/hollow_sphere

### CODE REVIEW

File: benchmarks/hollow_sphere/hollow_sphere.cc [hunk 1]
```diff
@@ -255,43 +255,35 @@ namespace aspect
...
+              out.densities[i] = density(pos);
+
+              out.specific_heat[i] = 0;
+              out.thermal_conductivities[i] = 0.0;
+              out.compressibilities[i] = 0;
+              out.thermal_expansion_coefficients[i] = 0;
+            }
+        }
+
+        virtual double density (const Point<dim> &position) const;
```
Comment (tjhei) [inline benchmarks/hollow_sphere/hollow_sphere.cc]: can you remove density?

**Comment by bangerth** [comment]
Out of curiosity, have you run the benchmark and compared the results with the one that were obtained previously?

**Comment by alarshi** [comment]
In this case, I compared only for the first time step.

**Comment by tjhei** [comment]
> Out of curiosity, have you run the benchmark and compared the results with the one that were obtained previously?

This file is used in the unit test, so the change should be okay.

**Comment by alarshi** [comment]
Modified density.

**Comment by tjhei** [comment]
Can you also remove the density() function please?

**Comment by bangerth** [comment]
Addresses #1538.

**Pull request merged**